### ISSUE TITLE
The bots use an old Firefox version for testing

### ISSUE THREAD

**Issue by Snuffleupagus** [mozilla/pdf.js#16904]
**Steps to reproduce the problem:**
Look at recent bot logs, for both Linux and Windows. Some recent examples (taken from PR #16899):
 - http://54.241.84.105:8877/5de7eb750fa6ace/output.txt
 - http://54.193.163.58:8877/4779353b9acdc6f/output.txt

**What is the expected behavior? (add screenshot)**
That *the latest* Firefox version is being used for testing, such that we're able to catch browser regressions.

**What went wrong? (add screenshot)**
The bots still use Firefox 115 during the reference tests, since the following user-agents are logged:
 - `User agent: Mozilla/5.0 (X11; Linux x86_64; rv:109.0) Gecko/20100101 Firefox/115.0`
 - `User agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:109.0) Gecko/20100101 Firefox/115.0`

Looking at the logs it does seem that Firefox 119 was correctly downloaded, however it's not actually being used.

*Please note:* I don't know exactly when this started, but I wonder if the `trimCache`-functionality (see PR #16519) isn't working as intended since older Firefox versions are apparently not being removed!?
As far I can tell, old Firefox versions not being removed seem to reproduce locally as well. @timvandermeij Would you have time to take a look?

In the short term, would it be possible to *manually* clear old Firefox versions from the Puppeteer-cache on the bots? @calixteman Do you have time to do that?

**Comment by calixteman** [comment]
I just removed the directories containing 115, 116, 117 and 118 on both bots.

**Comment by timvandermeij** [comment]
Interesting find; it might imply an upstream bug in Puppeteer. I'll have a look this weekend and try to find out what's happening.

**Comment by timvandermeij** [comment]
I have reproduced this problem locally and found that it's indeed an upstream bug in Puppeteer. Old Chrome revisions do get cleaned up correctly, but old Firefox releases don't, even though there is code to do so in the `trimCache` implementation upstream. However, the test that was added on the Puppeteer side only tests Chrome cleanup which is likely how the Firefox bug went unnoticed.

I'll try to debug the problem in Puppeteer and report a bug upstream.
